### DISCUSSION TITLE
Updating state in Google Home app

### DISCUSSION THREAD

**Discussion by mosaicws** [mikejac/node-red-contrib-google-smarthome#560] [Q&A] [answered]
I'm able to send `return {
    "payload": {
        "on": false
    }
};` to a Device node and update it's state but now I want to have that updated state shown in the Google Home app.

I've seen in the docs built into the Management node: 

`A topic of report_state will send an update of all states to Google. msg.payload is not used for anything. A topic of request_sync requests Google to sync to learn about new or changed devices. msg.payload is not used for anything. `

So I've created an inject node to test and set msg.topic to "resuest_sync" and also tried "report_state" to send these msg.topic to the Management node. When I change the device state i.e. set a Device node to "on" this isn't reflected in the Google Home app, even when I use the inject node to set a msg.topic to the above options to "sync" or "report_state". Interestingly, when I manually press and hold to open the device settings it updates at that point.

Whats the proper way of doing this? Ideally I want the lowest latency possible between setting state locally and it being reflected in the Google Home app. Can I feed the output of the Device node back into the Management node to initiate an update of state in the Google Home app?

**Comment by FireWizard52** [comment]
Hello.

You wrote:

> I'm able to send return { "payload": { "on": false } }; to a Device node and update it's state but now I want to have that updated state shown in the Google Home app.

It should do that. I use the same configuration and it works almost instantly
The only thing that is different, that I send beside `"on": false `also `"online": true`

See:
```
{
    msg.payload = {
        on: true,
        online: true
    }
}

return msg;
```

 

> Interestingly, when I manually press and hold to open the device settings it updates at that point.

What do you mean with that?

> Can I feed the output of the Device node back into the Management node to initiate an update of state in the Google Home app?

No, this should not be necessary. I never did that.

I'm not sure if  the `"online": true` will solve your issue, but my devices are always synchronized if operated from different controls.

Regards

**Reply by mosaicws** [reply]
That's strange it works for you instantly, in my Google Home app when I update the state of a Device node in Node red, the Google Home app on my phone doesn't reflect this change unless I press and hold the device entity: 
![Screenshot 2024-08-04 160557](https://github.com/user-attachments/assets/5e7d15e6-b074-4825-846a-6ff69cd4b96b)
![Screenshot_20240804-160252](https://github.com/user-attachments/assets/f2b5cf33-10c6-4b31-a4ec-f1f716f9c680)

So you can see that in Node Red, the "Light Node" is On but in the Google Home app, the same device is showing as Off.

If I press and hold the "Node" device icon in the Google Home app on my phone it will suddenly update to reflect the change in Node Red but this doesn't seem to be happening automatically. This is why I was wondering if it was necessary to initiate a manual sync/update but you say this shouldn't be needed.

I've tried `return {
    "payload": {
        "on": true,
        "online": true
    }
};` but have the same issue.

Even when I use an inject node to send msg.topic: "request_sync" to the Management node  the device in the Google Home app doesn't update (but again this shouldn't be needed?). Am I missing something or could there be a misconfiguration on my end?

EDIT: For anyone else having this problem it seems that the device needs to be added to a home/room before it will automatically update when the state is changed in Node Red. I added this to a "test" room and now it does seem to work as expected with state changes reflected after a second or so delay. In other words, if the device is in the "Linked to you" section only (at the bottom of the Rooms/Devices list in the Google Home app) it doesn't seem to update it's state automatically.

**Comment by FireWizard52** [comment, marked as answer]
Hello @mosaicws 

I have all my devices linked to a Room and so that it is probably the reason, I never met this issue. I have always a name in "Room Hint" in the configuration of the "Device" node. I have also to confess, that I do not use the Google Home app to control the devices very often.

I also know, that ypu have to refresh my phone screen after a state change.

But it's nice, that it works as desired now.

Regards

**Reply by mosaicws** [reply]
Thanks for your help and feedback - I'll do further testing but if "Room hint" automatically adds the device to a Room in Google Home then that would explain why you never had a problem. All good, now I just need to set up my other devices.

**Comment by Caprico85** [comment]
NIce to know that you got it working. I didn't know about the room assignment either. Same as FireWizard52, my devices are added to rooms and I rarely use the app anyway.

I'll add a note to the setup guide and the troubleshooting section about this.
